fluster/decoders/iso_mpeg2_video.py

```diff
@@ -0,0 +1,91 @@
+# Fluster - testing framework for decoders conformance
+# Copyright (C) 2024, Fluendo, S.A.
+#  Author: Rubén Sánchez <rsanchez@fluendo.com>, Fluendo, S.A.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public License
+# as published by the Free Software Foundation, either version 3
+# of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library. If not, see <https://www.gnu.org/licenses/>.
+import glob
+import os
+
+from fluster.codec import Codec, OutputFormat
+from fluster.decoder import Decoder, register_decoder
+from fluster.utils import _read_file_as_bytes, run_command
+
+
+@register_decoder
+class ISOMPEG2VDecoder(Decoder):
+    """ISO MPEG2 Video reference decoder implementation"""
+
+    name = "ISO-MPEG2-VIDEO"
+    description = "ISO MPEG2 Video reference decoder"
+    codec = Codec.MPEG2_VIDEO
+    binary = "mpeg2decode"
+    is_reference = True
+
+    def decode(
+        self,
+        input_filepath: str,
+        output_filepath: str,
+        output_format: OutputFormat,
+        timeout: int,
+        verbose: bool,
+        keep_files: bool,
+    ) -> str:
+        """Decodes input_filepath in output_filepath"""
+        base_dir = os.path.splitext(os.path.basename(os.path.normpath(output_filepath)))[0]
+        temp_dir = os.path.join(os.path.join(os.path.dirname(output_filepath), base_dir))
+        os.makedirs(temp_dir, exist_ok=True)
+        output_filepath += ".yuv"
+
+        run_command(
+            [self.binary, "-b", input_filepath, "-f", "-r", "-o0", os.path.join(temp_dir, "rec%d")],
+            timeout=timeout,
+            verbose=verbose,
+        )
+
+        merge_result = merge_yuv_files(temp_dir, output_filepath)
+
+        return merge_result
+
+
+def merge_yuv_files(input_dir: str, output_filepath: str) -> str:
+    """Merge YUV frames into an only raw .yuv file for mpeg2 video test suite"""
+    num_frames = len(glob.glob(os.path.join(input_dir, "rec*.Y")))
+
+    if num_frames == 0:
+        raise ValueError("No frames were decoded")
+
+    data = bytearray()
+
+    for frame_num in range(num_frames):
+        frame_name = f"rec{frame_num}"
+        y_file = os.path.join(input_dir, f"{frame_name}.Y")
+        u_file = os.path.join(input_dir, f"{frame_name}.U")
+        v_file = os.path.join(input_dir, f"{frame_name}.V")
+
+        if not (os.path.exists(y_file) and os.path.exists(u_file) and os.path.exists(v_file)):
+            print(f"Warning: Files for frame {frame_name} not found in {input_dir}")
+            continue
+
+        y_norm = _read_file_as_bytes(y_file)
```

Comment:
Instead of reading all the frames for all the components, start with the y plane, read in blocks of 1024 and write them to the output file, iterate until you have no more data to read and repeat for the rest of planes.